.github/workflows/README.md

```diff
@@ -2,12 +2,8 @@
 
 This collection of GitHub workflow definitions is designed for Continuous Integration (CI) and is organized as follows:
 
-- **correctness-linux.yaml**: Builds and runs the entire test suite on `ubuntu-latest` using both _gcc_ and _clang_ compilers. Both `Debug` and `Release` builds are tested, with `AddressSanitizer` and `UndefinedBehaviorSanitizer` enabled.
+- **correctness-linux.yaml**: Builds and runs the entire test suite on `ubuntu-latest` using both _gcc_ and _clang_ compilers. Both `Debug` and `Release` builds are tested, with `AddressSanitizer`, `LeakSanitizer`, and `UndefinedBehaviorSanitizer` enabled.  The CppCheck and ClangTidy static analysis tools are both enabled for all configurations.
 
-- **correctness-macos.yaml**: Builds and runs the entire test suite on `macos-latest` using the _AppleClang_ compiler. Similar to the Linux workflow, both `Debug` and `Release` builds are tested, with `AddressSanitizer` and `UndefinedBehaviorSanitizer` enabled.
+- **correctness-macos.yaml**: Builds and runs the entire test suite on `macos-latest` using the _clang_ compiler, both the default version provided by Apple and the latest version. Similar to the Linux workflow, both `Debug` and `Release` builds are tested, with `AddressSanitizer` and `UndefinedBehaviorSanitizer` enabled.  The CppCheck and ClangTidy static analysis tools are both enabled for all configurations.
```

Comment:
As I said in the description for the PR, g++ is actually forcing CI to use AppleClang.

More info:  AppleClang doesn't come with ClangTidy, so you have to build it yourself.  The easiest way to do this is to do `spack install llvm` and then `spack load llvm`.  However, loading llvm sets the default environment compiler to be the clang compiler that we've build instead of the one that comes with the system or is specified with the CI environment variables.  However, the combination of setting up the CI environment compiler variable to g++ and then passing it to cmake as the CMAKE_CXX_COMPILER allows us to use AppleClang.  The clang++ compiler option uses the clang++ that we just built so that we can have access to ClangTidy.  The run having a different name than what is actually going on is unfortunate.

Reply:
Thanks for the explanation - it makes sense.